tests/layer_tests/common/tf_layer_test_class.py

```diff
@@ -62,3 +66,134 @@ def get_framework_results(self, inputs_dict, model_path):
+                input_shapes = kwargs['input_shapes']
+                input_names = kwargs['input_names']
+                assert len(input_shapes) == len(input_names)
+                input_dict = {}
+                for input_name, input_shape in zip(input_names, input_shapes):
+                    input_dict[input_name] = input_shape
+                mo_params.update(dict(input=input_dict))
+            elif 'input_names' in kwargs:
+                mo_params.update(dict(input=kwargs['input_names']))
+            elif 'input_shapes' in kwargs:
+                mo_params.update(dict(input=kwargs['input_shapes']))
+
+        exit_code, stderr = generate_ir_python_api(**mo_params)
+        assert not exit_code, (
+            "IR generation failed with {} exit code: {}".format(exit_code, stderr))
+
+        path_to_xml = Path(temp_dir, 'model.xml')
+        path_to_bin = Path(temp_dir, 'model.bin')
+
+        config = None
+        # GPU default execution precision is FP16, so if we want to check FP32 inference
+        # we need to set explicit precision hint
+        if ie_device == 'GPU' and precision == 'FP32':
+            config = {'INFERENCE_PRECISION_HINT': 'f32'}
+
+        ie_engine = InferAPI(model=path_to_xml,
+                             weights=path_to_bin,
+                             device=ie_device,
+                             use_legacy_frontend=use_legacy_frontend)
+        # Prepare feed dict
+        if 'kwargs_to_prepare_input' in kwargs and kwargs['kwargs_to_prepare_input']:
+            inputs_dict = self._prepare_input(ie_engine.get_inputs_info(precision),
+                                              kwargs['kwargs_to_prepare_input'])
+        else:
+            inputs_dict = self._prepare_input(ie_engine.get_inputs_info(precision))
+
+        # Infer using OpenVINO
+        infer_res = ie_engine.infer(input_data=inputs_dict, infer_timeout=infer_timeout, config=config)
+
+        # Infer using the original framework (TensorFlow or TFLite)
+        fw_res = self.get_framework_results(inputs_dict=inputs_dict, model_path=model_path)
+
+        if 'custom_eps' in kwargs and kwargs['custom_eps'] is not None:
+            custom_eps = kwargs['custom_eps']
+        else:
+            if precision == 'FP32':
+                custom_eps = 1e-4
+            else:
+                custom_eps = 5e-2
+        # Compare Ie results with Framework results
+        assert self.compare_ie_results_with_framework(infer_res=infer_res, framework_res=fw_res,
+                                                      framework_eps=custom_eps), \
+            "Comparing with Framework failed: ie_res={}; framework_res={}.".format(infer_res,
+                                                                                   fw_res)
+
+    # Feed dict for each input is filled with random number.
+    # It is possible to redefine this function and generate your own input
+    def _prepare_input(self, inputs_dict):
```

Comment:
Do we need to redefine this method here? Looks identical to `_prepare_input` in the base class: https://github.com/openvinotoolkit/openvino/blob/fc391e4c85c294c14975d4afa6f3b1e414835da5/tests/layer_tests/common/layer_test_class.py#L160

Reply:
Let us have it identical. We will have new implementation in the future if needed.